activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -547,6 +547,80 @@ def check_constraints(table_name) # :nodoc:
           end
         end
 
+        # Returns an array of exclusion constraints for the given table.
+        # The exclusion constraints are represented as ExclusionConstraintDefinition objects.
+        def exclusion_constraints(table_name)
+          scope = quoted_scope(table_name)
+
+          exclusion_info = exec_query(<<-SQL, "SCHEMA")
+            SELECT conname, pg_get_constraintdef(c.oid) AS constraintdef
+            FROM pg_constraint c
+            JOIN pg_class t ON c.conrelid = t.oid
+            WHERE c.contype = 'x'
+              AND t.relname = #{scope[:name]}
```

Comment:
Would you consider to add schema name support as check constraints does via https://github.com/rails/rails/commit/cc8637dcef359b013689e7cee1aa0cdbdc52f545 ?

Reply:
Definitely! Just added it.